### ISSUE TITLE
Use memfd for file data

### ISSUE THREAD

**Issue by Xyene** [DMOJ/judge-server#835]
We can create a memfd pointing to decompressed problem data, and pass that off to the submissions.

Pros:

* no more piping necessary in judge; fewer context switches as a result
* submissions can seek the fd
* still no disk thrashing

Cons(?):

* 3.7+ kernel necessary
* extra memcpy

**Comment by quantum5** [comment]
Support for creation of memfd-backed files is implemented in #836, with unlinked temporary files as fallback on FreeBSD.

This should now be a matter of writing decompressed data into an `MemoryIO` instance and passing the `fd` to `TracedPopen`.

**Comment by Xyene** [comment]
Some stats from prod (numbers are the total number of context switches, summed across all test cases):

```
miracle-base-1: 1402
miracle-base-2: 1201
miracle-tip-1: 47
miracle-tip-2: 57
rmt-base-1: 65887
rmt-base-2: 65318
rmt-base-3: 61411
rmt-tip-1: 25833
rmt-tip-2: 25849
```

`miracle` refers to https://dmoj.ca/problem/miraclesort, `rmt` to https://dmoj.ca/problem/ccc17s5.

I expect we could decrease the number of RMT context switches by implementing writing submission output to memfds instead.
